### ISSUE TITLE
Issue with Spark action and Polynote 0.4.0

### ISSUE THREAD

**Issue by Hako93** [polynote/polynote#1168]
**Description**
When I launch every spark action with Polynote and Cluster mode, the code won't terminate and I have the following warning at the Polynote terminal :
 
[Timer-0] WARN org.apache.spark.scheduler.cluster.YarnScheduler - Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources.

The same code work well with Spark-shell or Jupyter in cluster mode and with Polynote with local mode.

**Reproduction**
launch Spark action like show, first, or collect.

- Polynote Version: 
   - Example: 0.4.0
- Operating System: 
  -  Ubuntu 18.04.5 LTS
- JVM Version:
  - Example: 11
- Python Version:
  - Example: 2.7.17
- How you installed Polynote: 
  - I downloaded this release : https://github.com/polynote/polynote/releases/download/0.4.0/polynote-dist.tar.gz
- The command you are using to start Polynote:
  - Example: `./polynote.py`
- Spark Version :
  - 3.1.1

**Comment by gogol123** [comment]
Yes same issue here, until yet didn’t find anything to bypass the issue
